src/util/triplea/MapXMLCreator/ImageScrollPanePanel.java

```diff
@@ -0,0 +1,309 @@
+	{
+		if (s_polygonsInvalid)
+		{
+			s_polygons.clear();
+			s_polygonsInvalid = false;
+		}
+		if (!s_polygons.isEmpty())
+			return;
+		File file = null;
+		if (MapXMLCreator.s_mapPolygonsFile == null)
+		{
+			if (MapXMLCreator.s_mapFolderLocation != null && MapXMLCreator.s_mapFolderLocation.exists())
+				file = new File(MapXMLCreator.s_mapFolderLocation, "polygons.txt");
+			if (file == null || !file.exists())
+				file = new File(MapXMLCreator.s_mapImageFile.getParent() + File.separator + "polygons.txt");
+		}
+		else
+		{
+			file = MapXMLCreator.s_mapPolygonsFile;
+		}
+		if (MapXMLCreator.s_mapPolygonsFile != null
+					|| file.exists()
+					&& JOptionPane.showConfirmDialog(new JPanel(), "A polygons.txt file was found in the map's folder, do you want to use the file to supply the territory shapes?", "File Suggestion",
+								1) == 0)
+		{
+			try
+			{
+				System.out.println("Load Polygons from " + file.getPath());
+				s_polygons = PointFileReaderWriter.readOneToManyPolygons(new FileInputStream(file.getPath()));
+			} catch (final IOException ex1)
+			{
+				System.out.println("Something wrong with your Polygons file");
+				ex1.printStackTrace();
+			}
+		}
+		else
+		{
+			try
+			{
+				System.out.println("Select the Polygons file");
+				final String polyPath = new FileOpen("Select A Polygon File", MapXMLCreator.s_mapFolderLocation, ".txt").getPathString();
+				if (polyPath != null)
+				{
+					System.out.println("Polygons : " + polyPath);
+					s_polygons = PointFileReaderWriter.readOneToManyPolygons(new FileInputStream(polyPath));
+				}
+				else
+				{
+					System.out.println("Polygons file not given. Will run regardless");
+				}
+			} catch (final IOException ex1)
+			{
+				System.out.println("Something wrong with your Polygons file");
+				ex1.printStackTrace();
+			}
+		}
+		MapXMLCreator.s_mapPolygonsFile = file;
+	}
+
+	protected static String findTerritoryName(final Point p, final Map<String, List<Polygon>> m_polygons)
+	{
+		String seaName = "unknown";
+		// try to find a land territory.
+		// sea zones often surround a land territory
+		for (final String name : m_polygons.keySet())
+		{
+			final Collection<Polygon> polygons = m_polygons.get(name);
+			for (final Polygon poly : polygons)
+			{
+				if (poly.contains(p))
+				{
+					if (name.endsWith("Sea Zone") || name.startsWith("Sea Zone"))
+					{
+						seaName = name;
+					}
+					else
+					{
+						return name;
+					}
+				}// if
+			}// while
+		}// while
+		return seaName;
+	}
+
+	private static Map<String, Point> loadCenters()
+	{
+		Map<String, Point> centers = new HashMap<String, Point>(); // hash map for center points
+		try
+		{
+			System.out.println("Load Centers from " + MapXMLCreator.s_mapCentersFile.getAbsolutePath());
+			final FileInputStream in = new FileInputStream(MapXMLCreator.s_mapCentersFile);
+			centers = PointFileReaderWriter.readOneToOne(in);
+		} catch (final FileNotFoundException ex)
+		{
+			ex.printStackTrace();

```

Comment:
I don't see the point, the idea is to have an empty centers map in case of an exception and that is what the method is doing. E.g. Paint should still happen.